### ISSUE TITLE
CUDA 3.0.2 triggers error in REPL

### ISSUE THREAD

**Issue by luraess** [omlins/ParallelStencil.jl#26]
CUDA v3.0.2 fix of
- Synchronize REPL expressions before returning ([#837](https://github.com/JuliaGPU/CUDA.jl/pull/837))

produces now an error within ParallelStencil when running code on the `Base.Threads` backend in the REPL when CUDA isn't functional (i.e. no `libcuda` is available).

[This `if` statement addition](https://github.com/JuliaGPU/CUDA.jl/blob/45a7217f4f7d3ee0caa07d4957d55421e92796ae/src/initialization.jl#L77) in CUDA.jl (src/initialization.jl) seems to be responsible for this.

Note that ParallelStencil has CUDA as dependency but won't use it unless activated in `@init_parallel_stencil()`  

Error stack trace:
```Julia-repl
ERROR: CUDA.jl did not successfully initialize, and is not usable.
If you did not see any other error message, try again in a new session
with the JULIA_DEBUG environment variable set to 'CUDA'.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] libcuda()
    @ CUDA ~/.julia/packages/CUDA/MpASK/src/initialization.jl:52
  [3] macro expansion
    @ ~/.julia/packages/CUDA/MpASK/lib/cudadrv/libcuda.jl:29 [inlined]
  [4] macro expansion
    @ ~/.julia/packages/CUDA/MpASK/lib/cudadrv/error.jl:94 [inlined]
  [5] cuDeviceGet
    @ ~/.julia/packages/CUDA/MpASK/lib/utils/call.jl:26 [inlined]
  [6] CuDevice
    @ ~/.julia/packages/CUDA/MpASK/lib/cudadrv/devices.jl:25 [inlined]
  [7] CUDA.TaskLocalState()
    @ CUDA ~/.julia/packages/CUDA/MpASK/src/state.jl:50
  [8] task_local_state!()
    @ CUDA ~/.julia/packages/CUDA/MpASK/src/state.jl:73
  [9] stream
    @ ~/.julia/packages/CUDA/MpASK/src/state.jl:419 [inlined]
 [10] synchronize()
    @ CUDA ~/.julia/packages/CUDA/MpASK/lib/cudadrv/stream.jl:117
 [11] top-level scope
    @ ~/.julia/packages/CUDA/MpASK/src/initialization.jl:83
```

**Comment by luraess** [comment]
It seems to be a pure CUDA.jl issue as reported in this [MVE](https://gist.github.com/luraess/764358200223712d804bbf3d08e14d0a)

Temporary fix is to downgrade to CUDA v3.0.1:

```Julia-repl
using Pkg
Pkg.add(Pkg.PackageSpec(;name="CUDA", version="3.0.1"))
```

**Comment by luraess** [comment]
https://github.com/JuliaGPU/CUDA.jl/pull/840 fixes this CUDA.jl issue, bumping CUDA to v3.0.3.
